haskell/private/cabal_wrapper.sh.tpl

```diff
@@ -12,15 +12,33 @@
 
 set -euo pipefail
 shopt -s nullglob
-execroot="$(pwd)"
 
-# Set these variables if unset.
-: ${LD_LIBRARY_PATH:=}
-: ${LIBRARY_PATH:=}
-export LD_LIBRARY_PATH=$execroot/${LD_LIBRARY_PATH//:/:$execroot/}
-export LIBRARY_PATH=$execroot/${LIBRARY_PATH//:/:$execroot/}
+# Poor man's realpath, because realpath is not available on macOS.
+function realpath()
+{
+    [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"
+}
+
+function canonicalize_path()
+{
+    new_path=""
+    while IFS=: read -r -d: entry
+    do
+	if [[ -n "$entry" ]]
+	then
+	    new_path="$new_path${new_path:+:}$(realpath "$entry")"
+	fi
+    done <<< "${1:-}:"
+    echo $new_path
+}
+
+# Remove any relative entries, because we'll be changing CWD shortly.
+LD_LIBRARY_PATH=$(canonicalize_path $LD_LIBRARY_PATH)
+LIBRARY_PATH=$(canonicalize_path $LIBRARY_PATH)
+PATH=$(canonicalize_path $PATH)
```

Comment:
I'm a bit surprised that we have to do the absolute name resolution in Bash. I would have thought Bazel would give us tools to do so.

Reply:
Any Bazel supplied absolute paths would wreak havoc on the action cache. Nix gets away with this because the store is always in the same location. But the Bazel cache and the action execution sandbox could be anywhere.

Reply:
Oh! I get it! It could be quite terrible for shared cache too.

That being said, these global names should not be a dependency nor an output of the target, so I'm not sure it would matter for this particular case.

Reply:
Right, I meant for any shared cache. In general there will be paths to inputs going in one or more of the above 3 environment variables.